nntrainer/tensor/cpu_backend/cblas_interface/cblas_interface.cpp

```diff
@@ -15,11 +15,30 @@
 #include <cblas_interface.h>
 
 namespace nntrainer {
+
+/**
+ * @brief Set openblas #thread.
+ * @param [in] num_threads Use BLAS_NUM_THREADS if < 0.
+ *                         Otherwise, directly call openblas API.
+ */
+void __openblas_set_num_threads(int num_threads) {
+  if (num_threads < 0) {
+  #ifdef BLAS_NUM_THREADS
+    openblas_set_num_threads(BLAS_NUM_THREADS);
+  #else
+    /// Without openblas_set_num_threads,
+    /// it's set std::thread::hardware_concurrency()
+    /// It can be too high especially when the given blas function is small
+    /// or if there are other threads already created (nntrainer thread pool)
+    /// With big-little & threadboost, hardware_concurrency might be not good.
+    /// @todo configure this! (4? num of big cores? ...)
+  #endif
+  } else {
+    openblas_set_num_threads(num_threads);
+  }
+}
 void __cblas_saxpy(const unsigned int N, const float alpha, const float *X,
                    const unsigned int incX, float *Y, const unsigned int incY) {
-#ifdef BLAS_NUM_THREADS
-  openblas_set_num_threads(BLAS_NUM_THREADS);
-#endif
```

Comment:
I'm not sure we have to check if the num thread is propery set in every unpack function. We could just delete the openblas_set_num_threads in every function. and we just set it when we create the Network.
But for now, It is ok. We could remove the __openblase_set_num_threads call in unpack function later.